src/analysisd/decoders/security_configuration_assessment.c

```diff
@@ -926,14 +968,21 @@ static void HandleScanInfo(Eventinfo *lf,int *socket,cJSON *event) {
                 description_db = description->valuestring;
             }
 
-            result_event = SavePolicyInfo(lf,socket,policy->valuestring,file->valuestring,policy_id->valuestring,description_db,references_db);
+            result_event = SavePolicyInfo(lf,socket,policy->valuestring,file->valuestring,policy_id->valuestring,description_db,references_db,hash_file->valuestring);
             if (result_event < 0)
             {
                 merror("Error storing scan policy monitoring information for agent %s", lf->agent_id);
             }
-            
             break;
         default:
+            os_calloc(OS_MAXSTR, sizeof(char), old_hash);
+            if(FindPolicySHA256(lf, policy_id->valuestring, socket, old_hash) == 0){
+                if(strcmp(hash_file->valuestring, old_hash)){
```

Comment:
Add here an informative message to tell the user that the policy file changed and the whole scan will be reported again from the agent. If not, we have no way to know what happened.